### PR TITLE
Switch from s3 to dynamodb & allow category update of a given description

### PR DESCRIPTION
**Pull request by weilu**
Note that the category update endpoint will remove any date found from the description before trying a substring/contains match with transactions in the db. This is to increase the likelihood of matching.

### CODE REVIEW

**Comment by weilu** [comment]
This also resolves #5

File: backend/app.py [hunk 1]
```diff
@@ -157,6 +172,51 @@ def upload():
...
+def batch_tx_writes(uuid, tx_df):
+    # ignore category as the same transaction may be classified differently by different models
+    content = tx_df.drop(columns=['category']).to_csv()
+    content_hash = hashlib.sha1(content.encode('utf-8')).hexdigest()
+    updated_at = str(datetime.utcnow())
+    requests = []
+    for index, row in tx_df.iterrows():
+        item = {
+          "uuid": {
+            "S": uuid
```
Comment (ghost) [inline backend/app.py:184]: So `S` is a string type, `N` is number? it's like the data type for each field.

File: backend/app.py [hunk 2]
```diff
@@ -170,56 +230,82 @@ def confirm():
...
+
+    return Response(body='Updated {} transactions'.format(len(items)), status_code=200)
+
+
+@app.route('/transactions', methods=['GET'], cors=True, authorizer=authorizer)
 def transactions():
     uuid = get_current_user_email()
-    files = s3.list_objects(Bucket=CSV_BUCKET, Prefix=uuid)['Contents']
-    df = s3_csvs_to_df(files)
+    response = dynamodb.query(**query_by_uuid_param(uuid))
```
Comment (ghost) [inline backend/app.py:287]: `query_by_uuid_param` return one dictionary why do we need to put `**` here? `**` is for allowing an arbitrary number of args right?

File: backend/app.py [hunk 3]
```diff
@@ -170,56 +230,82 @@ def confirm():
...
+    df = dynamodb_response_to_df(response)
     return dataframe_as_response(df, app.current_request.headers['accept'])
 
 
 @app.route('/refresh-model', methods=['GET'])
 def refresh_model():
     classifier, model_obj = get_model(CLASSIFIER_FILENAME)
+    # note: LastModified is based on when the model file is last written on s3
```
Comment (ghost) [inline backend/app.py:295]: Do we still have anything store on S3?

**Review by ghost — APPROVED** [review]

File: backend/app.py (hunk 1 above)
Comment (weilu) [reply backend/app.py:184]: Yup. It's dynamoDB's request payload format: https://boto3.readthedocs.io/en/latest/reference/services/dynamodb.html

File: backend/app.py (hunk 2 above)
Comment (weilu) [reply backend/app.py:287]: It's like a splat operation. You can think of it as converting a dictionary to key-value arguments.

File: backend/app.py (hunk 3 above)
Comment (weilu) [reply backend/app.py:295]: Yes, ML models are serialized and stored on S3 and the raw statement PDFs are also stored on S3. DynamoDB does support binary data type, but I think it's not necessary to move at this point. We can always move it later if necessary.

**Pull request merged**